community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageCursor.java

```diff
@@ -103,18 +104,85 @@ public final long getCurrentPageId()
     }
 
     /**
-     * NOTE: Must be called while holding the right translationTableLock.writeLock
-     * for the given translationTable!!!
-     * This method will release that write lock on the translation table as part
-     * of the page faulting!
+     * Pin the desired file page to this cursor, page faulting it into memory if it isn't there already.
+     * @param filePageId The file page id we want to pin this cursor to.
+     * @param exclusive 'true' if we will be taking an exclusive lock on the page as part of the pin.
+     * @throws IOException if anything goes wrong with the pin, most likely during a page fault.
      */
-    protected void pageFault(
-            long filePageId,
-            PrimitiveLongObjectMap<MuninnPage> translationTable,
-            StampedLock translationTableLock,
-            long ttlStamp,
-            PageSwapper swapper ) throws IOException
+    protected void pin( long filePageId, boolean exclusive ) throws IOException
+    {
+        PageSwapper swapper = pagedFile.swapper;
+        pinEvent = pagedFile.tracer.beginPin( exclusive, filePageId, swapper );
+        int chunkId = pagedFile.computeChunkId( filePageId );
+        // The chunkOffset is the addressing offset into the chunk array object for the relevant array slot. Using
+        // this, we can access the array slot with Unsafe.
+        long chunkOffset = pagedFile.computeChunkOffset( filePageId );
+        Object[][] tt = pagedFile.translationTable;
+        Object[] chunk;
+        if ( tt.length <= chunkId )
+        {
+            tt = pagedFile.expandCapacity( chunkId );
+        }
+        chunk = tt[chunkId];
```

Comment:
`chunk` could be declared on this line. The divided declaration and assignment is an unnecessary mental stumbling block when reading the code ("why is this separated? is is assigned in the if? nope. huh, ok, strange.")